lib/cext/ruby.h

```diff
@@ -450,6 +450,15 @@ void rb_num_zerodiv(void);
 #define NUM2SIZET(x) NUM2ULONG(x)
 #define NUM2SSIZET(x) NUM2LONG(x)
 
+#define LLONG_MIN_MINUS_ONE ((double)LLONG_MIN-1)
+#define LLONG_MAX_PLUS_ONE (2*(double)(LLONG_MAX/2+1))
+LONG_LONG rb_num2ll(VALUE val);
+#define NUM2LL(x) rb_num2ll(x)
+#define LLONG_MIN_MINUS_ONE_IS_LESS_THAN(n) \
+  (LLONG_MIN_MINUS_ONE == (double)LLONG_MIN ? \
+   LLONG_MIN <= (n): \
+   LLONG_MIN_MINUS_ONE < (n))
```

Comment:
These macros, except NUM2LL, should be part of ruby.c.
In MRI, they are defined in numeric.c.
Only public functions/macros from MRI include should be part of the `ruby.h` (and the needed `rb_tr*` debug functions).